projects/rocprofiler-sdk/source/lib/python/rocpd/libpyrocpd.cpp

```diff
@@ -469,12 +468,151 @@ PYBIND11_MODULE(libpyrocpd, pyrocpd)
+                            "    INNER JOIN `rocpd_string` SN ON SN.id = T.name_id "
+                            "    AND SN.guid = T.guid "
+                            "    INNER JOIN `rocpd_string` CN ON CN.id = E.category_id "
+                            "    AND CN.guid = T.guid "
+                            "    INNER JOIN `rocpd_info_pmc` PMC_I ON SN.string =  "
+                            "    CASE CN.string "
+                            "        WHEN 'amd_smi' THEN PMC_I.name || ' [' || "
+                            "           (SELECT type_index "
+                            "            FROM `rocpd_info_agent` A "
+                            "            WHERE A.id = PMC_I.agent_id) || ']' "
+                            "        ELSE  PMC_I.name  "
+                            "    END "
+                            "    AND PMC_I.guid = T.guid "
+                            "    INNER JOIN `rocpd_info_agent` A ON A.id = PMC_I.agent_id "
+                            "    AND A.guid = T.guid "
+                            "    LEFT OUTER JOIN `rocpd_info_thread` TH ON TH.id = T.tid "
+                            "    AND TH.guid = T.guid; ";
+
+                        sqlite3_stmt* _stmt_samples_temp;
+                        uint64_t      prepare_samples_status = sqlite3_prepare_v2(
+                            conn, create_samples_view.c_str(), -1, &_stmt_samples_temp, nullptr);
+                        if(prepare_samples_status != SQLITE_OK)
+                        {
+                            fprintf(stderr,
+                                    "Error preparing samples temp view: %s\n",
+                                    sqlite3_errmsg(conn));
+                            assert(false);
+                        }
+                        uint64_t step_samples_status = sqlite3_step(_stmt_samples_temp);
+                        if(step_samples_status != SQLITE_DONE)
+                        {
+                            fprintf(stderr,
+                                    "Error creating samples temp view: %s\n",
+                                    sqlite3_errmsg(conn));
+                            assert(false);
+                        }
+
                         auto samples = rocpd::sql_generator<rocpd::types::sample>{
-                            conn, select_guid_nid_pid("samples"), sample_order_by};
+                            conn, select_guid_nid_pid("samples_temp"), sample_order_by};
 
                         auto threads = rocpd::sql_generator<rocpd::types::thread>{
                             conn, select_guid_nid_pid("threads")};
 
+                        auto pmc_info = rocpd::sql_generator<rocpd::types::pmc_info>{
+                            conn, select_guid_nid_pid("pmc_info")};
+
+                        const std::string create_pmc_event_view =
+                            "CREATE TEMP VIEW "
+                            "    `pmc_events_temp` AS "
```

Comment:
```suggestion
                            "    `pmc_events_temp` AS "
```
Don't use temp... maybe refer to "schema_3_0" instead

Reply:
Done